File: GDTMobApp/Release/GDTAds/GDTUnifiedNativeAd/Cell/UnifiedNativePortraitCollectionViewCell.m
//
//  UnifiedNativePortraitCollectionViewCell.m
//  GDTMobApp
//
//  Created by royqpwang on 2019/5/16.
//  Copyright © 2019 Tencent. All rights reserved.
//

#import "UnifiedNativePortraitCollectionViewCell.h"
#import "GDTLogoView.h"


@interface UnifiedNativePortraitCollectionViewCell()

@property (nonatomic, strong) UIImageView *imageView;

@end

@implementation UnifiedNativePortraitCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self initViews];
    }
    return self;
}

- (void)initViews
{
    self.backgroundColor = [UIColor blueColor];
    [self addSubview:self.imageView];
}

- (void)setupWithUnifiedNativeAdDataObject:(GDTUnifiedNativeAdDataObject *)nativeAdDataObject
{
    NSURL *imageURL = [NSURL URLWithString:nativeAdDataObject.imageUrl];
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        NSData *imageData = [NSData dataWithContentsOfURL:imageURL];
        dispatch_async(dispatch_get_main_queue(), ^{
            self.imageView.image = [UIImage imageWithData:imageData];
        });
    });
}

- (void)prepareForReuse
{
    [super prepareForReuse];
    self.imageView.image = nil;
}

#pragma mark - property getter
- (UIImageView *)imageView
{
    if (!_imageView) {
        _imageView.translatesAutoresizingMaskIntoConstraints = YES;
        _imageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        _imageView = [[UIImageView alloc] initWithFrame:self.bounds];
        _imageView.accessibilityIdentifier = @"imageView_id";
    }
    return _imageView;
}


@end

File: GDTMobSample-Swift/GDTAdTypes/UnifiedNativeAdViewController/cell/UnifiedNativePortraitCollectionViewCell.swift
//
//  UnifiedNativePortraitCollectionViewCell.swift
//  GDTMobSample-Swift
//
//  Created by 胡城阳 on 2019/10/29.
//  Copyright © 2019 Tencent. All rights reserved.
//

import UIKit

class UnifiedNativePortraitCollectionViewCell: UICollectionViewCell {
    
    var theImageView:UIImageView?
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.blue
        theImageView = UIImageView(frame: self.bounds)
        theImageView?.translatesAutoresizingMaskIntoConstraints = true
        theImageView?.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.flexibleWidth.rawValue | UIViewAutoresizing.flexibleHeight.rawValue)
        self.addSubview(theImageView!)
    }
    
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func setupWithUnifiedNativeAdDataObject(nativeAdDataObject: GDTUnifiedNativeAdDataObject){
        let imageURL:NSURL = NSURL.init(string: nativeAdDataObject.imageUrl)!
        DispatchQueue.global(qos: .default).async {
            let imageData = NSData.init(contentsOf: imageURL as URL)
            DispatchQueue.main.async {
                self.theImageView!.image = UIImage(data: imageData! as Data)
            }
        }
    }
    
    override func prepareForReuse() {
        super.prepareForReuse()
        self.theImageView?.image = nil
    }
}
